src/secrets-kubernetes/src/lib.rs

```diff
@@ -100,12 +132,35 @@ impl SecretsController for KubernetesSecretsController {
 
         self.secret_api
             .patch(
-                &SECRET_NAME,
+                &self.config.user_defined_secret,
                 &PatchParams::apply(FIELD_MANAGER).force(),
                 &Patch::Apply(secret),
             )
             .await?;
 
+        self.trigger_resync().await?;
+
+        // guarantee that all new secrets are reflected on our local filesystem
+        let secrets_storage_path =
+            PathBuf::from(self.config.user_defined_secret_mount_path.clone());
+        for op in ops.iter() {
+            match op {
+                SecretOp::Ensure { id, .. } => {
+                    let file_path = secrets_storage_path.join(format!("{}", id));
+                    Retry::default()
+                        .max_duration(Duration::from_secs(POLL_TIMEOUT))
+                        .retry(|_| {
```

Comment:
This is an async method so it should use the `retry_async` variant to avoid synchronously blocking in a Tokio task.

Reply:
`Path::exists` is not async.

Reply:
Yes, so you'll need to use an equivalent async method from Tokio instead of `Path::exists`. I believe you have to check the error return from https://docs.rs/tokio/latest/tokio/fs/fn.metadata.html, but perhaps there is something better.